version: "3.6"

services:
  # https://stackoverflow.com/questions/71370929/mongo-express-is-not-trying-to-reconnect-after-first-attempt
  db:
    image: mongo:6.0
    restart: always
    environment:
      MONGO_INITDB_ROOT_USERNAME: root
      MONGO_INITDB_ROOT_PASSWORD: password
    volumes:
      - ./data/db:/data/db
      - ./data/config:/data/configdb
    command: mongod --port 27017 --auth

  mongo-express:
    image: mongo-express:1.0.0-alpha.4
    restart: always
    ports:
      - 8081:8081
    environment:
      ME_CONFIG_BASICAUTH_USERNAME: test # Display authenticate popup
      ME_CONFIG_BASICAUTH_PASSWORD: test
      ME_CONFIG_MONGODB_ADMINUSERNAME: root
      ME_CONFIG_MONGODB_ADMINPASSWORD: password
      ME_CONFIG_MONGODB_SERVER: db
    depends_on:
      - db

  server:
    tty: true # Don't understand
    stdin_open: true # Don't understand
    build: ./
    command: npm start
    volumes:
      - ./server:/app
      - ./server/node_modules:/app/node_modules
    depends_on:
      - mongo-express
    ports:
      - 3000:3000

  # web:
  #   tty: true
  #   stdin_open: true
  #   build: ./
  #   command: npm start
  #   volumes:
  #     - ./web:/app
  #     - './web/node_modules:/app/node_modules'
  #   depends_on:
  #     - server
  #   ports:
  #     - 5000:3000
